nucypher/blockchain/eth/actors.py

```diff
@@ -656,3 +696,342 @@ def request_evaluation(self, evidence):
+
+    @property
+    def funding_eth(self) -> Decimal:
+        ethers = self.blockchain.client.get_balance(self.funding_account)
+        return ethers
+
+    #
+    # Staking Utilities
+    #
+
+    def read_onchain_stakes(self, account: str = None) -> None:
+        if account:
+            accounts = [account]
+        else:
+            accounts = self.__accounts
+
+        for account in accounts:
+            stakes = list(self.staking_agent.get_all_stakes(staker_address=account))
+            if stakes:
+                staker = Staker(is_me=True, checksum_address=account, blockchain=self.blockchain)
+                self.__stakers[staker] = staker.stakes
+
+    @property
+    def total_stake(self) -> NU:
+        total = sum(staker.locked_tokens() for staker in self.stakers)
+        return total
+
+    @property
+    def stakers(self) -> List[Staker]:
+        return list(self.__stakers.keys())
+
+    @property
+    def stakes(self) -> list:
+        payload = list()
+        for staker in self.__stakers:
+            payload.extend(staker.stakes)
+        return payload
+
+    @property
+    def account_balances(self) -> dict:
+        balances = dict()
+        for account in self.__accounts:
+            funds = {'ETH': self.blockchain.client.get_balance(account),
+                     'NU': self.token_agent.get_balance(account)}
+            balances.update({account: funds})
+        return balances
+
+    @property
+    def staker_balances(self) -> dict:
+        balances = dict()
+        for staker in self.stakers:
+            staker_funds = {'ETH': staker.eth_balance, 'NU': staker.token_balance}
+            balances = {staker.checksum_address: staker_funds}
+        return balances
+
+    def __serialize_stakers(self) -> list:
+        payload = list()
+        for staker in self.stakers:
+            payload.append(staker.to_dict())
+        return payload
+
+    def get_active_staker(self, address: str) -> Staker:
+        self.read_onchain_stakes(account=address)
+        for staker in self.__stakers:
+            if staker.checksum_address == address:
+                return staker
+        else:
+            raise self.NoStakes(f"{address} does not have any stakes.")
+
+    def __create_staker(self, password: str = None) -> Staker:
+        """Create a new account and return it as a Staker instance."""
+        if self.funding_power.device:
+            # TODO: More formal check for wallet here?
+            # With devices, the last account is is always an unsed one.
+            new_account = self.blockchain.client.accounts[-1]
+        else:
+            if not password:
+                raise self.ConfigurationError("No password supplied to create new staking account.")
+            new_account = self.blockchain.client.new_account(password=password)
+        self.__accounts.append(new_account)
+        staker = Staker(blockchain=self.blockchain, is_me=True, checksum_address=new_account)
```

Comment:
Shouldn't this staker be added to `self.__stakers` too?

Reply:
Also fixed in `68426a32076f60be232a5b50b458f5da56d06f31`

Reply:
to me, these little fixes are a reason not to do the whole
```
@property
def thing(self)

set_thing(self, thing)

get_thing(self, thing)

self.__thing
```

just another bug to have

Reply:
Heh - Just generally prefer statelessness?

Reply:
python objects already have a thing called `__dict__` and methods called `setattr` and `getattr` that help manage state.

self.thing = new_value
print (self.thing)

that is state enough no?
I guess if we want to do some kind of validation on the attribute being set it makes sense, but we're not really.

Also, there is nothing stopping a developer from directly modifying these attributes anyway.

Reply:
Sometime I want to perform some logic during `object.attr = value`.  In python this can be implemented with the `@property.getter` and `@property.setter` methods, but I typically prefer to use named methods for this sort of operation.